src/main/java/online/book/store/controller/BookController.java

```diff
@@ -28,30 +29,35 @@ public class BookController {
 
     private final BookService bookService;
 
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
     public List<BookDto> findAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
     @GetMapping("/{id}")
     @Operation(summary = "Get book by ID", description = "Get the existing book by ID")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBook(id);
     }
 
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
Try to be consistent when choosing to use `hasRole` or `hasAuthority`. In this app this is basically the same, but in bigger projects it usually serves different thigs.